src/vs/platform/dialogs/common/dialogs.ts

```diff
@@ -18,100 +18,166 @@ import { isLinux, isMacintosh, isWindows } from 'vs/base/common/platform';
 import { IProductService } from 'vs/platform/product/common/productService';
 import { deepClone } from 'vs/base/common/objects';
 
-export interface FileFilter {
-	readonly extensions: string[];
-	readonly name: string;
+export interface IDialogArgs {
+	readonly confirmArgs?: IConfirmDialogArgs;
+	readonly inputArgs?: IInputDialogArgs;
+	readonly promptArgs?: IPromptDialogArgs;
 }
 
-export type DialogType = 'none' | 'info' | 'error' | 'question' | 'warning';
+export interface IBaseDialogOptions {
+	readonly type?: Severity | DialogType;
 
-export interface ICheckbox {
-	readonly label: string;
-	readonly checked?: boolean;
+	readonly title?: string;
+	readonly message: string;
+	readonly detail?: string;
+
+	readonly checkbox?: ICheckbox;
+
+	/**
+	 * Allows to enforce use of custom dialog even in native environments.
```

Comment:
Maybe a hint when one might want to do this? You mentioned elsewhere that native prompts do not support checkboxes. Does that mean you have to turn this on when using a checkbox above, or do other, non-prompt native dialogs support checkboxes?

Reply:
No, you do not have to set `custom` explicitly to do anything, we make sure that for example for `input` it will be custom because there are no Electron APIs for this.

But when you want to render rich markdown content or use a custom icon, you need to use `custom`. Feel free to add that in this PR.